libraries/rib-debug-utils/src/main/java/com/badoo/ribs/core/plugin/utils/debug/AbstractDebugControls.kt

```diff
@@ -0,0 +1,67 @@
+package com.badoo.ribs.core.plugin.utils.debug
+
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.badoo.mobile.rib.base.R
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.customisation.inflate
+import com.badoo.ribs.core.plugin.NodeAware
+import com.badoo.ribs.core.plugin.NodeAwareImpl
+import com.badoo.ribs.core.plugin.RibAware
+import com.badoo.ribs.core.plugin.RibAwareImpl
+import com.badoo.ribs.core.plugin.ViewLifecycleAware
+
+@SuppressWarnings("LongParameterList")
+abstract class AbstractDebugControls<T : Rib> internal constructor(
```

Comment:
maybe it would be better to define an interface for debug controls and use composition to build it instead of inheriting from `AbstractDebugControls`?

Reply:
Different params are needed for the two use-cases (represented as child classes), but under the hood it's common functionality, and they need to work together. I don't see how an interface would help here, since they're plugins, we don't interact with them from the outside.